templates/admin/controls/facets-settings.php

```diff
@@ -0,0 +1,186 @@
+<?php
+/**
+ * @var \Swiftype\SiteSearch\Wordpress\Admin\Page $this
+ */
+?>
+
+<div class="card">
+    <h3><?php echo __('Configure facets'); ?></h3>
```

Comment:
Probably should have a description of what a facet is somewhere.

Reply:
Plan to add the following text in a tootltip:

> Facets allow users to narrow down search results by applying filters based on the post fields.

About facet rendering, you have to add the following code somewhere in your theme `search.php` file (`wp-content/themes/twentynineteen/search.php`for the default theme) :

```php
<?= swiftype_render_facets(); ?>
```

From what I know, there is nothing we can do to add it by default.
I plan to add the following message in the facet config box:

> If facets are not displayed in your search results make sure you have called the `swiftype_render_facets` theme function in the `search.php` file of your theme.

Would this works for you ?

Reply:
That should work. I basically was just looking for some sort of instructions on how to add facets after I've configured them.

Reply:
Alright, dropped that snippet in my template and now it's 👌 . However, none of my Category facets show up. Does this work for you?

Reply:
Actually, it's almost as if the facets are always one change behind the actual posts.

For instance.

I add the "Tests" category to a post and update. That category does not show up in facets.
I then remove the "Tests" category from a post and update. That category now shows up in facets, even though there are now now records tagged as "Tests"

Reply:
Looks to me like some sort of issue syncing up with Swiftype.

Reply:
Will investigate the category lagging.
Think the same applies to tags.

Reply:
Have resolved lagging for categories and tags when saving the post from admin (5b43074).
Was due to a change in WP 5.x (we need to use new hook to handle these changes that are not processed at the same time than the main post save).

Reply:
Have added help to the facet config (b68d60d):

![Uploading Capture d’écran 2019-02-28 à 09.06.01.png…]()

All subject in this topics are now OK for me.
You can review again and send me your feedback.